### PR TITLE
[Artist] Show Venice Biennale banner if a participant.

### PR DESCRIPTION
**Pull request by alloy**
Closes https://github.com/artsy/collector-experience/issues/242

### CODE REVIEW

File: desktop/apps/artist/components/current_item/index.jade [hunk 1]
```diff
@@ -4,6 +4,6 @@ a.current-item(href= currentItem.href)
       img.current-item__image(src= currentItem.imageUrl)
   .current-item__right
     .current-item__label.avant-garde-s-headline= currentItem.heading
-    .current-item__name= currentItem.name
+    .current-item__name(style= currentItem.name && currentItem.detail && "white-space:nowrap;")= currentItem.name
```
Comment (alloy) [inline desktop/apps/artist/components/current_item/index.jade]: i.e. allow wrapping if only a `name` is specified.

File: desktop/apps/artist/components/current_venice_feature/index.coffee [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+cacheKey = "venice-2017"
+
+fetchVeniceArtists = ->
+  Q.promise (resolve, reject) ->
+    cache.get cacheKey, (err, cachedData) ->
+      return reject(err) if err
+      return resolve(JSON.parse(cachedData)) if cachedData
+
+      request
+        .get "http://files.artsy.net/force/venice_2017_artists.json"
```
Comment (alloy) [inline desktop/apps/artist/components/current_venice_feature/index.coffee]: Simply uploaded with https://vanity.artsy.net, is that ok?

File: desktop/apps/artist/components/current_venice_feature/index.coffee [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+module.exports = (artist, fetch = fetchVeniceArtists) ->
+  fetch()
+    .then (veniceArtists) ->
+      for veniceArtist in veniceArtists
+        if veniceArtist.slug == artist.id
+          return {
+            type: 'eoy',
+            href: '/venice-biennale',
+            imageUrl: 'http://files.artsy.net/images/venice_2017.jpg',
```
Comment (alloy) [inline desktop/apps/artist/components/current_venice_feature/index.coffee]: Same

File: desktop/apps/artist/routes.coffee [hunk 4]
```diff
@@ -35,20 +36,23 @@ sd = require('sharify').data
       return res.redirect(artist.href) unless(_.find nav.sections(), slug: tab) or artist.counts.artworks is 0
 
       if (req.params.tab? or artist.href is res.locals.sd.CURRENT_PATH)
-          currentItem = currentShowAuction(artist)
+        currentVeniceFeature(artist)
+          .then (veniceFeature) ->
+            currentItem = veniceFeature or currentShowAuction(artist)
+            console.log(currentItem)
```
Comment (alloy) [inline desktop/apps/artist/routes.coffee]: Whoops

Comment (mzikherman) [inline desktop/apps/artist/routes.coffee]: 🌴

File: desktop/apps/artist/test/components/current_venice_feature/index.coffee [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+describe "current venice feature", ->
+  fetch = () -> Q.resolved [
+    { slug: "featured", featured: true },
+    { slug: "participant", featured: false }
+  ]
+
+  it "renders nothing if the given artist is not in the list of artists", ->
+    currentItem({ id: "not-featured" }, fetch).then (result) ->
+      (result == undefined).should.equal true
```
Comment (mzikherman) [inline desktop/apps/artist/test/components/current_venice_feature/index.coffee]: nice! I'd been using `_.isUndefined(...).should.be.ok()` (and then import `isUndefined` from lodash), but this is 👍 too!

File: desktop/apps/artist/test/components/current_venice_feature/index.coffee [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+  it "renders an artist featured in editorial content", ->
+    currentItem({ id: "featured", name: "Banksy" }, fetch).then (result) ->
+      result.heading.should.match(/INSIDE/)
```
Comment (mzikherman) [inline desktop/apps/artist/test/components/current_venice_feature/index.coffee]: also cool! I'd used `containEql` for a substring comparison, looks like this is 👍

File: desktop/apps/artist/components/current_venice_feature/index.coffee (hunk 2 above)
Comment (mzikherman) [inline desktop/apps/artist/components/current_venice_feature/index.coffee]: Yea, seems 👍 to me. I guess it doesn't matter that the file isn't SSL, but we should maybe look into that at some point on vanity's S3 folder (or cloudfront distro or whatever).

File: desktop/apps/artist/components/current_venice_feature/index.coffee (hunk 3 above)
Comment (mzikherman) [inline desktop/apps/artist/components/current_venice_feature/index.coffee]: wonder if we'd get that 'mixed content' warning on browsers b/c of the SSL comment ^

File: desktop/apps/artist/components/current_venice_feature/index.coffee [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+            heading: if veniceArtist.featured then 'INSIDE THE BIENNALE' else '57TH VENICE BIENNALE PARTICIPANT',
+            name: if veniceArtist.featured
+                    name = artist.name + (if artist.name.slice(-1) == "s" then "’" else "’s")
```
Comment (mzikherman) [inline desktop/apps/artist/components/current_venice_feature/index.coffee]: neat

File: desktop/apps/artist/test/components/current_venice_feature/index.coffee (hunk 5 above)
Comment (alloy) [inline desktop/apps/artist/test/components/current_venice_feature/index.coffee]: I like that much better 👍

**Comment by alloy** [comment]
@mzikherman Addressed all feedback.

File: desktop/apps/artist/components/current_venice_feature/index.coffee [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+cacheKey = "venice-2017"
+
+fetchVeniceArtists = ->
+  Q.promise (resolve, reject) ->
+    cache.get cacheKey, (err, cachedData) ->
+      return reject(err) if err
+      return resolve(JSON.parse(cachedData)) if cachedData
+
+      request
+        .get "https://artsy-vanity-files-production.s3.amazonaws.com/force/venice_2017_artists.json"
```
Comment (mzikherman) [inline desktop/apps/artist/components/current_venice_feature/index.coffee]: 🔒

**Pull request merged**